pkg/plugins/plugin.go

```diff
@@ -14,3 +20,42 @@ type VendorPlugin interface {
 	// Apply config change
 	Apply() error
 }
+
+type HostHelpersInterface interface {
```

Comment:
i think each plugin should require the interfaces that are relevant.

Reply:
I was thinking to do it but here we just past a pointer to the object with all the interfaces I think it makes the code much more cleaner and if the plugin doesn't need all the function that is fine it's there

Reply:
i think each plugin would eventually need 2 or 3 (some zero) interfaces.
and it would be clearer from code POV what each plugin ingests as interfaces
(e.g mlx plugin just needs mellanox interface and kernel/driver interface.

Reply:
the idea for this interface was to not check the ability or create objects for the vendor plugins

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/e95012cb1907ddf0ff1225f091e6dd1962cda676/pkg/daemon/plugin.go#L19

if every vendor plugin requires different interfaces we are not going to be able to use the map for the vendors no?

I can send the hostHelper to all the vendors and inside every vendor package will take a subset (I think it will work) is that a good solution?

Reply:
yes its ok.